### DISCUSSION TITLE
OpenJDK support of liboqs

### DISCUSSION THREAD

**Discussion by greengalaxy2016** [open-quantum-safe/liboqs#1610] [Ideas]
is there any plan to support liboqs in OpenJDK?

**Comment by dstebila** [comment]
We don't have any plans in this direction. But if you have ideas for what that would entail and are interested to work on it, we'd be very happy to hear it.

**Reply by greengalaxy2016** [reply]
Basically, our team is experimenting E2EE (end-to-end-encryption, an asymmetric encryption) using liboqs. Our application is java based.  liboqs algorithms are not supported in OpenJDK crypto package. So it might worth considering to plug KEM algorithms into OpenJDK. 
p.s. there is a similar discussion regarding CRYSTAL-Kyber for encryption/decryption.

**Reply by greengalaxy2016** [reply]
Bouncy Castle Project provides a light weighted cryptography Java API which allows using Kyber for symmetric encryption.
